pkg/operator/defaultstorageclass/controller.go

```diff
@@ -160,6 +160,9 @@ func (c *Controller) syncStorageClass() error {
 }
 
 func newStorageClassForCluster(infrastructure *configv1.Infrastructure) (*storagev1.StorageClass, error) {
+	if infrastructure.Status.PlatformStatus == nil {
+		return nil, unsupportedPlatformError
+	}
```

Comment:
Good catch!

Just a nit: I'd suggest moving this check right after `infrastructure` is fetched.

Reply:
I moved this up to `syncStorageClass()` and included a comment about why we're performing this check.